daemon/api_quota_test.go

```diff
@@ -0,0 +1,239 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2021 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package daemon_test
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"github.com/snapcore/snapd/daemon"
+	"github.com/snapcore/snapd/gadget/quantity"
+	"github.com/snapcore/snapd/overlord/servicestate"
+	"github.com/snapcore/snapd/overlord/state"
+	"github.com/snapcore/snapd/snap/quota"
+	"gopkg.in/check.v1"
+)
+
+var _ = check.Suite(&apiQuotaSuite{})
+
+type apiQuotaSuite struct {
+	apiBaseSuite
+}
+
+func (s *apiQuotaSuite) SetUpTest(c *check.C) {
+	s.apiBaseSuite.SetUpTest(c)
+	s.daemon(c)
+}
+
+func mockQuotas(st *state.State, c *check.C) {
+	grp0, err := quota.NewGroup("foo", 9000)
+	c.Assert(err, check.IsNil)
+	grp1, err := grp0.NewSubGroup("bar", 1000)
+	c.Assert(err, check.IsNil)
+	grp2, err := quota.NewGroup("baz", 2000)
+	c.Assert(err, check.IsNil)
+	c.Assert(servicestate.UpdateQuotas(st, grp0, grp1, grp2), check.IsNil)
```

Comment:
Maybe a small todo that want to use servicestate.CreateQuota() here instead? Given the discussion in https://github.com/snapcore/snapd/pull/10211#discussion_r622910239 it's likely we will stop export this. But for now it's fine of course until the dust has settled in 10211 :)